jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/FloydWarshallShortestPaths.java

```diff
@@ -60,12 +63,33 @@
     public FloydWarshallShortestPaths(Graph<V, E> graph)
     {
         super(graph);
+
+        /*
+         * Sort vertices by degree in ascending order and index them. Also compute the minimum
+         * vertex which has degree at least one and at least two.
+         */
         this.vertices = new ArrayList<>(graph.vertexSet());
+        Collections.sort(
+            vertices, new VertexDegreeComparator<>(graph, VertexDegreeComparator.Order.ASCENDING));
+        this.degrees = new ArrayList<>();
         this.vertexIndices = new HashMap<>(this.vertices.size());
         int i = 0;
+        int minDegreeOne = vertices.size();
+        int minDegreeTwo = vertices.size();
         for (V vertex : vertices) {
-            vertexIndices.put(vertex, i++);
+            vertexIndices.put(vertex, i);
+            int degree = graph.degreeOf(vertex);
+            degrees.add(degree);
+            if (degree > 0 && i < minDegreeOne) {
```

Comment:
I am sure that it won't change anything but you can rewrite it with at most 3 comparisons:

```
if (degree > 1){
      if (i < minDegreeOne){
          minDegreeOne = i;
      }
      if (i < minDegreeTwo){
          minDegreeTwo = i;
      }
}
else if (degree == 1 && i < minDegreeOne){
        minDegreeOne = i;
}
```